### PR TITLE
Bug fixes in base workflows and improved data-exploration UI 

### PR DESCRIPTION
**Pull request by alivinco**

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,12 @@ Changes are grouped as follows:
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [0.14.1] - 28-06-23
```
Comment (doctrino) [inline docs/CHANGELOG.md:17]:
```suggestion
## [0.14.1] - 30-06-23
```

File: cognite/neat/explorer/explorer.py [hunk 2]
```diff
@@ -310,7 +311,7 @@ def execute_rule(request: RuleRequest):
 @app.get("/api/object-properties")
 def get_object_properties(reference: str, graph_name: str = "source", workflow_name: str = "default"):
     logging.info(f"Querying object-properties from {graph_name} :")
-    query = f"SELECT ?property ?value WHERE {{ <{reference}> ?property ?value }}"
+    query = f"SELECT ?property ?value WHERE {{ <{reference}> ?property ?value }} ORDER BY ?property"
```
Comment (doctrino) [inline cognite/neat/explorer/explorer.py:314]: To ensure results are deterministic?

File: cognite/neat/core/extractors/rdf_to_assets.py [hunk 3]
```diff
@@ -526,7 +526,16 @@ def _asset2dict(asset: Asset) -> dict:
...
+    result = set()
+    if labels is None:
+        return result
+    for label in labels:
+        if "externalId" in label:
+            result.add(label["externalId"])
+        else:
+            logging.warning(f"Label {label} does not have externalId")
+    return result
+    # return {label["externalId"] for label in labels}
```
Comment (doctrino) [inline cognite/neat/core/extractors/rdf_to_assets.py:538]:
```suggestion
```

File: cognite/neat/explorer/explorer.py [hunk 4]
```diff
@@ -357,41 +358,101 @@ def get_nodes_and_edges(request: NodesAndEdgesRequest):
...
-        ?dst_object_ref  rdf:type ?dst_object_class .\
-        }} "
-    logging.info(f"Nodes query : {nodes_query}")
-    logging.info(f"Edges query : {edges_query}")
-    nodes_result = get_data_from_graph(nodes_query, request.graph_name, request.workflow_name)
-    edges_result = get_data_from_graph(edges_query, request.graph_name, request.workflow_name)
+    nodes_result = {}
+    edges_result = {}
+    query = ""
+    elapsed_time_sec = 0
```
Comment (doctrino) [inline cognite/neat/explorer/explorer.py:364]: These are not used (they are overwritten further down)
```suggestion
```

File: cognite/neat/explorer/explorer.py [hunk 5]
```diff
@@ -357,41 +358,101 @@ def get_nodes_and_edges(request: NodesAndEdgesRequest):
...
+            edges_result = [
+                {
+                    "src_object_ref": v[rdflib.Variable("src_object_ref")],
+                    "dst_object_ref": v[rdflib.Variable("dst_object_ref")],
+                }
+                for v in mixed_result["rows"]
+                if rdflib.Variable("src_object_ref") in v and rdflib.Variable("dst_object_ref") in v
+            ]
+        except Exception as e:
+            logging.error(f"Error while parsing edges : {e}")
```
Comment (doctrino) [inline cognite/neat/explorer/explorer.py:406]: Not sure if this makes sense, but should these be a helper functions? Something like `parse_node_results(mixed_results)` and `parse_edge_results(mixed_results)`.

**Review by doctrino — APPROVED** [review]
I tried to run `neat` but cannot get up the graph visualization, however, this is a problem on the `main` branch as well. Except for that just very minor comments.

File: cognite/neat/explorer/explorer.py (hunk 2 above)
Comment (alivinco) [reply cognite/neat/explorer/explorer.py:314]: YEs

File: cognite/neat/core/extractors/rdf_to_assets.py (hunk 3 above)
Comment (alivinco) [reply cognite/neat/core/extractors/rdf_to_assets.py:538]: yes

File: cognite/neat/explorer/explorer.py (hunk 4 above)
Comment (alivinco) [reply cognite/neat/explorer/explorer.py:364]: That's inherited from other languages where variables must be initialised first

File: cognite/neat/explorer/explorer.py (hunk 5 above)
Comment (alivinco) [reply cognite/neat/explorer/explorer.py:406]: Good point

**Pull request merged**